### PR TITLE
 add items get api

### PR DESCRIPTION
**Pull request by festinalente91**
Add Items get api

@fguy  @shberry16 @ovekyc @elegant651 @HewonJeong

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,7 +76,6 @@
     "browserify": "^13.1.0",
     "enzyme": "^2.4.1",
     "eslint-plugin-react": "^6.1.2",
-    "gulp": "^3.9.1",
```
Comment (ovekyc) [inline package.json:79]: bcz of duplicated both save and dev dependency

File: src/server/items/controllers.js [hunk 2]
```diff
@@ -2,18 +2,56 @@ import db from '../database';
 
 export default {
   getAll: (req, res, cb) => {
-    const result = {};
```
Comment (ovekyc) [inline src/server/items/controllers.js:5]: How about declare just const item = []
and req.send({ items: items });

File: src/server/items/controllers.js [hunk 3]
```diff
@@ -2,18 +2,56 @@ import db from '../database';
...
+        })
+        return cb(err);
+      }
+      
+      res.status(200);
+      res.send(value);
+      return cb();
     });
   },
-
```
Comment (ovekyc) [inline src/server/items/controllers.js:16]: please new line

File: src/server/items/controllers.js [hunk 4]
```diff
@@ -2,18 +2,56 @@ import db from '../database';
...
+    })
+    .on('close', () => {
+      if(result.items.length !== 0 ) {
+        res.send(result);
+        cb();
+      }
+    });
+  },
+  getById: (req, res, cb) => {
+    const key = 'item' + req.params.id;
```
Comment (fguy) [inline src/server/items/controllers.js]: Use template here:
const {id} = req.params;
const key = `item${id}`

File: src/server/items/controllers.js [hunk 5]
```diff
@@ -2,18 +2,56 @@ import db from '../database';
...
+    db.get(key, (err, value) => {
+      if(err){
+        if(err.notFound) {
+          res.status(400);
+          res.send({
+            error: "Item was not found."
```
Comment (fguy) [inline src/server/items/controllers.js]: Use single quotes for string literal. This must be caught by lint.

File: package.json (hunk 1 above)
Comment (festinalente91) [reply package.json:79]: right :)

File: src/server/items/controllers.js (hunk 2 above)
Comment (festinalente91) [reply src/server/items/controllers.js:5]: that would be nice

File: package.json (hunk 1 above)
Comment (festinalente91) [inline package.json]: oh, Trailing spaces not allowed by lint

File: src/server/items/controllers.js (hunk 3 above)
Comment (festinalente91) [reply src/server/items/controllers.js:16]: oh, Trailing spaces not allowed by lint

File: src/server/items/controllers.js (hunk 4 above)
Comment (ovekyc) [reply src/server/items/controllers.js]: baeutiful

**Comment by ovekyc** [comment]
LGTM

File: src/server/items/controllers.js [hunk 6]
```diff
@@ -2,18 +2,54 @@ import db from '../database';
...
+      }
+      res.status(500);
+      res.send({
+        error: 'database error'
+      });
+      return cb();
+    })
+    .on('close', () => {
+      if (items.length !== 0) {
+        res.send({items: items});
```
Comment (fguy) [inline src/server/items/controllers.js]: nit; short hand of {items: items} is {items}. This must be caught by lint.

**Comment by fguy** [comment]
LGTM besides of a nit comment. Please make sure your code doesn't occur lint warnings.

**Comment by fguy** [comment]
LGTM

**Pull request merged**